src/test/java/org/sagebionetworks/bridge/redis/RedisLockTest.java

```diff
@@ -0,0 +1,79 @@
+package org.sagebionetworks.bridge.redis;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.junit.Test;
+import org.mockito.InOrder;
+import org.sagebionetworks.bridge.lock.LockNotAvailableException;
+
+public class RedisLockTest {
+
+    @Test
+    public void testAcquireLock() {
+        JedisOps ops = mock(JedisOps.class);
+        // 1 -- indicates lock acquired
+        when(ops.setnx(eq("key"), anyString())).thenReturn(1L);
+        RedisLock lock = new RedisLock(ops);
+        lock.acquireLock("key", 10);
+        InOrder inOrder = inOrder(ops);
+        inOrder.verify(ops, times(1)).setnx(eq("key"), anyString());
+        inOrder.verify(ops, times(1)).expire("key", 10);
+    }
+
+    @Test
+    public void testAcquireLockWithLockNotAvailableException() {
+        JedisOps ops = mock(JedisOps.class);
+        // 0 -- indicates lock acquiring failed
+        when(ops.setnx(eq("key"), anyString())).thenReturn(0L);
+        when(ops.ttl("key")).thenReturn(30L);
+        RedisLock lock = new RedisLock(ops);
+        try {
+            lock.acquireLock("key", 10);
+        } catch (LockNotAvailableException e) {
+            assertTrue("LockNotAvailableException expected", true);
+        }
```

Comment:
This is actually not an effective way to test that the exception is thrown. If the exception is not thrown, the catch block is never executed, so the assertTrue() will never fire.

You should instead use a pattern like
try {
  lock.acquireLock("key", 10);
  fail("expected exception");
} catch (LockNotAvailableException e) {
  // stuff
}

I frequently like to extend this to
Exception thrownEx = null;
try {
  lock.acquireLock("key", 10);
  fail("expected exception");
} catch (LockNotAvailableException e) {
  thrownEx = e;
}
assertNotNull(thrownEx)

as a double-check that we did indeed throw an exception.